matching.go

```diff
@@ -285,39 +285,64 @@ func componentUniqueEmailsAndNames(graph *simple.UndirectedGraph, n simplegraph.
 }
 
 // SetPrimaryName sets people primary name to the most frequent name of the person's identity
-func SetPrimaryName(people People, nameFreqs map[string]int) {
+func SetPrimaryName(people People, nameFreqs map[string]*Frequency, minRecentCount int) {
+
 	for _, p := range people {
-		maxFreq := 0
-		primaryName := ""
+		recentMaxFreq := 0
+		totalMaxFreq := 0
+		sumRecentCount := 0
+		recentPrimaryName := ""
+		totalPrimaryName := ""
 		for _, name := range p.NamesWithRepos {
```

Comment:
Note: we should ignore the repos for this analysis. I am not sure if we do this or not here...

Reply:
Repos are not used in any way in the primaries detection